transaction.go

```diff
@@ -290,6 +291,97 @@ func (t *Transaction) SignalUpdatesApplied() error {
 	return nil
 }
 
+// append is a helper function to append updates to a transaction on which
+// SignalSetupComplete hasn't been called yet
+func (t *Transaction) append(updates []Update, done chan error) {
+	defer close(done)
+
+	// If there is nothing to append we are done
+	if len(updates) == 0 {
+		return
+	}
+
+	// Make sure that the initialization finished
+	err := <-t.initComplete
+	if err != nil {
+		done <- err
+		return
+	}
+
+	// Append the updates to the transaction
+	t.Updates = append(t.Updates, updates...)
```

Comment:
probably shouldn't modify `t` while there's still a chance that appending the update could fail, right?

Reply:
You are right :+1:

Reply:
Should we be enforcing this rule programmatically? IE, naively, by setting a flag while an update could fail and panicking if updates is appended before the flag is unset.